File: C++/1742. Maximum Number of Balls in a Box.cpp
class Solution {
public:
    int countBalls(int l, int h) {
        map<int,int>m;
        int a=0;
        for(int i=l;i<=h;i++){
            int s=0,x=i;
            while(x!=0){
                s+=x%10;
                x/=10;
            }
            m[s]++;
            a=max(m[s],a);
        }
        return a;
    }
};

File: JAVA/1742. Maximum Number of Balls in a Box.java
class Solution {
    public int countBalls(int l, int h) {
        Map<Integer, Integer> m = new HashMap<>();
        int a = 0;
        for (int i = l; i <= h; i++) {
            int s = 0, x = i;
            while (x != 0) {
                s += x % 10;
                x /= 10;
            }
            m.put(s, m.getOrDefault(s, 0) + 1);
            a = Math.max(m.get(s), a);
        }
        return a;
    }
}